test/common/http/match_delegate/config_test.cc

```diff
@@ -315,6 +315,63 @@ TEST(DelegatingFilterTest, MatchTreeSkipActionDecodingHeaders) {
   delegating_filter->decodeComplete();
 }
 
+// This function creates the matcher tree only with on_no_match set. So it can not be used with
+// on_match test cases.
```

Comment:
I would just remove the second part, it's not clear what that's referring to without digging into the rest of the test code